File: events/2018/01/01.rs
aoc::main!();

fn preprocessing(input: &str) -> Vec<i32> {
    input
    .lines()
    .map(|change| change.parse().unwrap())
    .collect_vec()
}

fn part_1(changes: Vec<i32>) -> i32 {
    changes
    .iter()
    .sum()
}

fn part_2(changes: Vec<i32>) -> i32 {
    changes
    .iter()
    .cycle()
    .scan(0, |frequency, &change| {
        *frequency += change;
        Some(*frequency)})
    .duplicates()
    .nth(0)
    .unwrap()
}

File: events/2018/01/01.py
def preprocessing(file):
    return list(int(x) for x in file.replace(', ', '\n').splitlines())

def solver(changes): 
    yield sum(changes)
    frequencies = set()
    frequency   = 0
    index       = 0
    while (frequency not in frequencies):
        if len(frequencies) > 1_000_000:
            yield None
            return
        frequencies.add(frequency)
        frequency += changes[index]
        index = (index + 1) % len(changes)
    yield frequency